### PR TITLE
Add Bandwidth Controls for Entitlement Tokens

### PR DESCRIPTION
**Pull request by apoclyps**
### What's Changed

Implements Bandwidth Controls for Entitlement Tokens

### Example Usage 

```bash
cloudsmith entitlements restrict example/testing/1xYSdAxlIqIV \
        --limit-bandwidth=1 \
        --limit-bandwidth-unit=gigabyte \
        --limit-num-clients=10 \
        --limit-num-downloads=1000 \
        --limit-package-query="package-darwin-amd64 OR package-windows-amd64 OR package-linux-amd64"  \
        --limit-path-query=tag:latest \
        --limit-date-range-from=2020-01-01T00:00:00Z \
        --limit-date-range-to=2077-01-01T00:00:00Z \
        --refresh-token=daily
```

Screenshot 2020-09-25 at 14 59 47

### CODE REVIEW

**Comment by lskillen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- cloudsmith_cli/cli/commands/entitlements.py  12
         

Clones added
============
- cloudsmith_cli/cli/commands/entitlements.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cloudsmith/cloudsmith-cli/pullRequest?prid=6237120&bid=20321285)

**Comment by lskillen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- cloudsmith_cli/cli/commands/entitlements.py  12
         

Clones added
============
- cloudsmith_cli/cli/commands/entitlements.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cloudsmith/cloudsmith-cli/pullRequest?prid=6237120&bid=20321285)

**Comment by lskillen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- cloudsmith_cli/cli/commands/entitlements.py  12
         

Clones added
============
- cloudsmith_cli/cli/commands/entitlements.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cloudsmith/cloudsmith-cli/pullRequest?prid=6237120&bid=20321285)

**Comment by lskillen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- cloudsmith_cli/cli/commands/entitlements.py  12
         

Clones added
============
- cloudsmith_cli/cli/commands/entitlements.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Cloudsmith/cloudsmith-cli/pullRequest?prid=6237120&bid=20321285)

File: cloudsmith_cli/cli/validators.py [hunk 1]
```diff
@@ -176,3 +176,63 @@ def validate_optional_timestamp(ctx, param, value):
...
+        "Annual",
+    ]
+
+    if value:
+        for period in periods:
+            if value.lower() == period.lower():
+                return period
+
+        raise click.BadParameter(
+            "The refresh token period must be one of the allow values "
```
Comment (MrTam) [inline cloudsmith_cli/cli/validators.py:232]:
```suggestion
            "The refresh token period must be one of the allowed values "
```

File: cloudsmith_cli/cli/validators.py [hunk 2]
```diff
@@ -176,3 +176,63 @@ def validate_optional_timestamp(ctx, param, value):
...
+        "Yottabyte",
+    ]
+
+    if value:
+        for unit in units:
+            if value.lower() == unit.lower():
+                return unit
+
+        raise click.BadParameter(
+            "Bandwidth values must be one of the allow values "
```
Comment (MrTam) [inline cloudsmith_cli/cli/validators.py:202]:
```suggestion
            "Bandwidth unit must be one of the allowed values "
```

**Review by MrTam — APPROVED** [review]
Some small suggestions to wording, but otherwise, 👍

**Pull request merged**